lms/djangoapps/mobile_api/testutils.py

```diff
@@ -123,8 +141,197 @@ def test_other_user(self):
+            namespace_choices.get('ENTRANCE_EXAM'),
+            self.course.id
+        )
+        milestone = {  # pylint: disable=attribute-defined-outside-init
+            'name': 'Test Milestone',
+            'namespace': milestone_namespace,
+            'description': 'Entrance Exam for TestMobileAPIMilestones',
+        }
+        milestone_relationship_types = get_milestone_relationship_types()
+        milestone = add_milestone(milestone)  # pylint: disable=attribute-defined-outside-init
+
+        self.course.entrance_exam_minimum_score_pct = 0.50
+        self.course.entrance_exam_id = unicode(self.entrance_exam.location)
+        modulestore().update_item(self.course, self.user.id)
+
+        # Add the exam
+        add_course_milestone(
+            unicode(self.course.id),
+            milestone_relationship_types['REQUIRES'],
+            milestone
+        )
+        add_course_content_milestone(
+            unicode(self.course.id),
+            unicode(self.entrance_exam.location),
+            milestone_relationship_types['FULFILLS'],
+            milestone
+        )
+
+    def _add_prerequisite_course(self):
+        """ Helper method to set up the prerequisite course """
+        self.prereq_course = CourseFactory.create()  # pylint: disable=attribute-defined-outside-init
+        add_prerequisite_course(self.course.id, self.prereq_course.id)
+
+    def _pass_entrance_exam(self):
+        """ Helper function to pass the entrance exam """
+        # set up the request for exam functions
+        request = RequestFactory()  # pylint: disable=attribute-defined-outside-init
+        request.user = self.user
+        request.COOKIES = {}
+        request.META = {}
+        request.is_secure = lambda: True
+        request.get_host = lambda: "edx.org"
+        request.method = 'GET'
+
+        # pylint: disable=maybe-no-member,no-member
+        grade_dict = {'value': 1, 'max_value': 1, 'user_id': self.user.id}
+        field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.user,
+            self.course,
+            depth=2
+        )
+        # pylint: disable=protected-access
+        module = get_module(
+            self.user,
+            request,
+            self.problem_1.scope_ids.usage_id,
+            field_data_cache,
+        )._xmodule
+        module.system.publish(self.problem_1, 'grade', grade_dict)
+
+    @patch.dict('django.conf.settings.FEATURES', {
+        'ENABLE_PREREQUISITE_COURSES': True,
+        'MILESTONES_APP': True,
+        'ENTRANCE_EXAMS': True
+    })
+    def test_feature_flags(self):
+        """
+        Tests when feature flags are set/unset, content is gated appropriately
+        """
+        self._add_prerequisite_course()
+        self._add_entrance_exam()
+        self.init_course_access()
+        if self.ALLOW_ACCESS_TO_MILESTONE_COURSE:
+            self.api_response(expected_response_code=None)
```

Comment:
Why do you have to pass `None` for `expected_response_code`?  Does it not return 200 in the success case?